src/optimizer/loadOptimizers.ts

```diff
@@ -39,17 +39,23 @@ export async function loadOptimizers(
   optimizers.forEach(opt => {
     nodes.push(opt.name);
     opt.runAfter?.forEach(dependsOn => {
-      if (optimizers.some(o => o.name === dependsOn)) {
-        edges.push([opt.name, dependsOn]);
+      const preceedingOptimizer = optimizers.find(o =>
+        dependsOn instanceof RegExp ? dependsOn.test(o.name) : dependsOn === o.name
+      );
+      if (preceedingOptimizer) {
+        edges.push([opt.name, preceedingOptimizer.name]);
       } else {
         logger.error(
           `The ${opt.name} optimizer specifies an unknown optimizer in runAfter: ${dependsOn}`
         );
       }
```

Comment:
I don't think this handling of `runAfter` totally works for `RegEx`. If the `runAfter` is a `RegEx`, then there may only be one expression in the array of `runAfter`. I think this code will only find one `preceedingOptimizer` that matches the `runAfter`, not all possible ones.

Reply:
You're right! Commit `8b519d4` switches this `find` to a `filter`, so it should organize the optimizers correctly now.